### ISSUE TITLE
[BUG] [monitoring v2] GKE hosted cluster monitoring does not connect to grafana or prometheus - connection refused

### ISSUE THREAD

**Issue by ronhorton** [rancher/rancher#40341]
**Rancher Server Setup**
- Rancher version: `v2.6-head commit id b465433`
- Installation option (Docker install/Helm Chart): `HELM`
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): `GKE`
- Proxy/Cert Details:

**Information about the Cluster**
- Kubernetes version: `1.24`
- Cluster Type (Local/Downstream): `hosted google gke cluster`
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider):

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom)
  - If custom, define the set of permissions: `admin`

**Describe the bug**
installed rancher-monitoring:100.2.0+up40.1.2; grafana dashboard has no data; connection refused; no prometheus links are available on monitoring page

**To Reproduce**
1. import gke hosted cluster
2. install rancher-monitoring:100.2.0+up40.1.2
3. navigate to cluster > monitoring > Grafana

**Result**
No data; connection refused; no prometheus links available

**Expected Result**
dashboards load and show data; prometheus links lead to appropriate pages

**Screenshots**
<img width="1438" alt="2023-01-26_17-02-14" src="https://user-images.githubusercontent.com/89212856/214961183-0422e018-1c6e-4c91-ad00-49621074ba23.png">
<img width="1429" alt="2023-01-26_17-03-43" src="https://user-images.githubusercontent.com/89212856/214961204-a593fd0b-63b4-4c83-92f3-b156ee8167a1.png">

**Additional context**

**Comment by aiyengar2** [comment]
@ronhorton I took a look at your setup and I think the root cause of the issue seen here might be your node machine type of `e2-medium (Efficient Instance, 2 vCPU (1/2 shared physical core) and 4 GB RAM)`.

**TLDR: you may need bigger instances to test.**

---

If you look at the reason why Prometheus isn't up, it seems to be a capacity issue.

![Screenshot 2023-01-26 at 5 26 23 PM](https://user-images.githubusercontent.com/13127443/214988860-b3d8ebf2-858c-4ccd-938c-44456f94f1d3.png)

Rancher Monitoring is configured to require a minimum of `750 mCPUs` (i.e. `0.75 CPUs`) and a maximum of `1000 mCPUs` (i.e. `1 CPU`), as seen below:

![Screenshot 2023-01-26 at 5 27 25 PM](https://user-images.githubusercontent.com/13127443/214989081-72a561b7-cf6d-4fd7-a3ad-baaaba0a8a60.png)

However, since none of your nodes have the capacity to offer the guaranteed limit of `1000 mCPUs` (since that's exactly 1 core; the max they seem to be able to offer is `0.94` cores), none of them are eligible for Prometheus to be deployed on.

To test this, I edited the values.yaml to remove both `prometheus.prometheusSpec.resources.requests` and `prometheus.prometheusSpec.resources.limits` (by setting `prometheus.prometheusSpec.resources={}`); this resulted in a successful deployment.

To be clear, **removing the limits and requests is not recommended**; you need bigger nodes for Monitoring V2.

**Comment by ronhorton** [comment]
**PASS** 
Verified in `v2.6-head commit ID da13242`

1. created gke cluster with 8 cpus per node
2. installed rancher-monitoring:100.2.0+up40.1.2
3. installed rancher-alerting-drivers:100.0.4

configured and received alerts
as indicated, this was a resource issue to begin with and wasn't a bug.
